bundles/automation/org.eclipse.smarthome.automation.core/src/main/java/org/eclipse/smarthome/automation/core/internal/RuleRegistryImpl.java

```diff
@@ -475,206 +453,116 @@ public Boolean isEnabled(String ruleUID) {
-                    rule.getConfiguration(), null, rule.getVisibility());
-            validateConfiguration(r1);
-            r1.setName(rule.getName());
-            r1.setTags(rule.getTags());
-            r1.setDescription(rule.getDescription());
-
-            return r1;
+    @Override
+    protected void addProvider(Provider<Rule> provider) {
+        super.addProvider(provider);
+        Collection<Rule> rules = new CopyOnWriteArraySet<Rule>(elementMap.get(provider));
+        for (Rule rule : rules) {
+            update(provider, rule);
         }
     }
 
-    public void templateUpdated(Collection<Template> templates) {
-        for (Template template : templates) {
-            String templateUID = template.getUID();
-            Set<String> rules = null;
-            synchronized (this) {
-                rules = mapTemplateToRules.get(templateUID);
-            }
-            if (rules != null) {
-                for (String rUID : rules) {
-                    RuleStatus ruleStatus = getStatusInfo(rUID).getStatus();
-                    if (ruleStatus == RuleStatus.NOT_INITIALIZED) {
-                        Rule oldRule, newRule;
-                        if ((oldRule = managedProvider.get(rUID)) != null) {
-                            // The rule belongs to managed provider
-                            newRule = resolveTemplate(oldRule);
-                            update(newRule);
-                        } else {
-                            // the rule is coming from read only provider and must not be stored
-                            oldRule = get(rUID);
-                            newRule = resolveTemplate(oldRule);
-                            onUpdateElement(oldRule, newRule);
-                        }
-                    }
-                }
-            }
+    @Override
+    public void added(Provider<Rule> provider, Rule element) {
+        Rule ruleWithUID = element;
+        if (element.getUID() == null) {
+            String rUID = ruleEngine.getUniqueId();
+            ruleWithUID = initRuleId(rUID, element);
         }
+        super.added(provider, ruleWithUID);
+        update(provider, ruleWithUID);
     }
 
-    protected void validateConfiguration(Rule r) {
-        List<ConfigDescriptionParameter> configDescriptions = r.getConfigurationDescriptions();
-        Configuration moduleConfiguration = r.getConfiguration();
-        Map<String, Object> configuration = moduleConfiguration.getProperties();
-        if (configuration != null) {
-            validateConfiguration(configDescriptions, new HashMap<String, Object>(configuration));
-            handleModuleConfigReferences(r.getTriggers(), configuration);
-            handleModuleConfigReferences(r.getConditions(), configuration);
-            handleModuleConfigReferences(r.getActions(), configuration);
+    private void update(Provider<Rule> provider, Rule rule) {
```

Comment:
Rename this method to updateRuleByTemplate, because update is not obvious what it is doing.